NSS-Spontaneous-Spoonbills/Sprint3
# Js payment form

## Link to Ticket

Autolink format: hashtag and number, ex: #1094

Closes #4

## Description of Proposed Changes
1. When the user is authenticated, they will be able to view their payment types via the navigation. 
*will be dropdown
1. In addition, you can add a payment and view it's details. 
*currently only shows the id #s for everything.

## Steps to Test
1. Have virtual machine running
1. Fetch and check out to branch
1. Remake migrations
1. Run server
1. Register two **NEW** users via the registration page
1. Open DB Browser
1. Execute payment type sql script below
1. In browser, ADD payment type
1. UPDATE payment type
1. LITERALLY TRY AND BREAK THIS THING, I DARE YOU
1. Give me the :shipit: :shipit: 

```
INSERT INTO website_payment_type VALUES (null, 'Visa');
INSERT INTO website_payment_type VALUES (null, 'Mastercard');
INSERT INTO website_payment_type VALUES (null, 'Discover');
INSERT INTO website_payment_type VALUES (null, 'Venmo');
INSERT INTO website_payment_type VALUES (null, 'Paypal');
```

```
git fetch --all
git checkout js_payment_form
```

## Impacted Areas in Application
1. urls.py
1. __init__py (forms & views)
1. templates/
1. views/

**New Files**:
```
	website/forms/payment_new_form.py
	website/templates/account/payment_option_new.html

```
**Modified Files**:
```
	modified:   website/forms/__init__.py
	modified:   website/forms/payment_option_form.py
	modified:   website/models/Payment_Type_Model.py
	modified:   website/templates/account/payment_option_detail.html
	modified:   website/templates/account/payment_option_list.html
	modified:   website/templates/account/payment_option_update.html
	modified:   website/urls.py
	modified:   website/views/__init__.py
	modified:   website/views/payment_option_view.py

```
**Deleted Files**:
```
       NONE

```
## Mentions @username

Tag users that need to review this code

## Testing


## Deploy Notes

Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
1. Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
